alacritty_terminal/src/config/mod.rs

```diff
@@ -134,8 +139,8 @@ pub struct Config {
     alt_send_esc: DefaultTrueBool,
 
     /// Shell startup directory
-    #[serde(default, deserialize_with = "failure_default")]
-    working_directory: WorkingDirectory,
+    #[serde(default, deserialize_with = "or_none")]
```

Comment:
I think we can do better than naming this `or_none`, which doesn't actually say a lot about what it does.

Something like `explicit_option` or `allow_none_option` might be more descriptive here. That should make it clear that we're only deserializing options here, not just any value or none.